core/src/main/java/io/grpc/internal/ServiceConfigUtil.java

```diff
@@ -411,101 +455,101 @@ public static String getStickinessMetadataKeyFromServiceConfig(
    * Gets a list from an object for the given key.
    */
   @SuppressWarnings("unchecked")
-  static List<Object> getList(Map<String, Object> obj, String key) {
+  static List<Object> getList(Map<String, Object> obj, String key) throws Exception {
     assert obj.containsKey(key);
     Object value = checkNotNull(obj.get(key), "no such key %s", key);
     if (value instanceof List) {
       return (List<Object>) value;
     }
-    throw new ClassCastException(
```

Comment:
ClassCastException was right before.  These getter methods imply a specific type, which ClassCastException is for.

Reply:
Done.